### DISCUSSION TITLE
Restarting Explorer resets Taskbars position

### DISCUSSION THREAD

**Discussion by ItsKunkey** [valinet/ExplorerPatcher#181] [Q&A] [answered]
Having the taskbar with the system tray on secondary monitor will reset to the original position on main monitor when restarting the computer or explorer.

is there a way to make it locked in position when using "lock all taskbars"?

OS Build : 22000.282

**Comment by valinet** [comment, marked as answer]
https://github.com/valinet/ExplorerPatcher/issues/171
https://github.com/valinet/ExplorerPatcher/issues/14#issuecomment-936412362

Third time asked. tl;dr Not a bug, it's by design of Explorer. I am looking into providing a solution for it, I see it is in very high demand apparently.

**Reply by ItsKunkey** [reply]
Thank you for your work , I guess its really good option for people who game on their main monitor and want to check on time/date and the tray icons "discord for example" on the secondary monitor.

**Reply by valinet** [reply]
Yeah, I understood also from the other thread. I am giving it a shot, I hope to come back with something. Thanks.

**Reply by valinet** [reply]
Okay, I have something. Can you help test this, please? There is a new setting in "Advanced": "Always show main taskbar on this monitor". How you use that is the following: 

1. Make sure the multi monitor taskbar is disabled (i.e. have only one taskbar, the one which resets to primary monitor currently).
2. Position the taskbar on the edge you want it to be, but put it on the *primary monitor*.
3. Choose a number from the list, the one you think the desired monitor is.
4. Click the "Restart File Explorer" link and watch where the taskbar is positioned. If it is right, then that's it, if not, repeat steps 3-4 until you get it right. (note: if a File Explorer window opens when restarting, close that and click "Restart File Explorer" again).

[dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7400407/dxgi.zip)

**Reply by ItsKunkey** [reply]
Thank you for your time , I disabled Multi monitor taskbar and I put the taskbar on bottom edge , picked all the numbers and It still on the primary monitor and the secondary is blank.

**Reply by valinet** [reply]
Apparently it does not work when primary is right, secondary left. I will come back with a fix, hopefully.

**Reply by ItsKunkey** [reply]
this is how the monitor is setup atm idk if would help and again thank you so much

![image](https://user-images.githubusercontent.com/92997392/138505314-e3dd873c-b0b3-46ed-99f8-03211a7ba1bc.png)

**Reply by valinet** [reply]
Same as mine and same as his: https://github.com/valinet/ExplorerPatcher/issues/171. let's all talk there. What resolution?
